### PR TITLE
Add Tests [minor]

### PR DESCRIPTION
**Pull request by ric-evans**
Add tests, with a focus on integration tests

### CODE REVIEW

File: .github/workflows/wipac-cicd.yml [hunk 1]
```diff
@@ -36,44 +43,67 @@ jobs:
...
+          python-version: ${{ matrix.version }}
+      - name: Integrate
         run: |
+          echo "--------------------------------------------------------------"
+          echo "starting broker..."
+
+          # Pulsar
+          if [ -z "${{ matrix.broker }}" ] || [ "${{ matrix.broker }}" = "pulsar" ]; then
+            wget https://raw.githubusercontent.com/WIPACrepo/MQClient/master/resources/docker-pulsar.sh
+            chmod +x ./docker-pulsar.sh
```
Comment (dsschult) [inline .github/workflows/wipac-cicd.yml:69]: Instead of doing a wget, should this be a second repo checkout?  You can checkout into a subdir.

**Review by dsschult — COMMENTED** [review]
Generally good, just the one comment

File: .github/workflows/wipac-cicd.yml (hunk 1 above)
Comment (ric-evans) [reply .github/workflows/wipac-cicd.yml:69]: Oh, that's pretty cool. I didn't know that was possible

File: .github/workflows/wipac-cicd.yml [hunk 2]
```diff
@@ -36,44 +43,64 @@ jobs:
...
+          elif [ "${{ matrix.broker }}" = "nats" ]; then
+            ./MQClient/resources/local-nats-js.sh
+          fi
+
+          echo "--------------------------------------------------------------"
+          echo "waiting for broker..."
+          sleep 15
+
+          echo "--------------------------------------------------------------"
+          echo "running tets..."
```
[the remark this answers is not in the document]
Comment (ric-evans) [reply .github/workflows/wipac-cicd.yml]: :+1:

File: .github/workflows/wipac-cicd.yml [hunk 3]
```diff
@@ -36,44 +43,65 @@ jobs:
...
-          pip install .[nats]
-          # pytest -vvvv
+          if [ -z "${{ matrix.broker }}" ]; then
+            export BROKER_CLIENT=pulsar
+            pip install .[test]
+          else
+            export BROKER_CLIENT=${{ matrix.broker }}
+            pip install .[test,${{ matrix.broker }}]
+          fi
+          pytest -vvv
```
[the remark this answers is not in the document]
Comment (dsschult) [reply .github/workflows/wipac-cicd.yml]: the reason for running `python -m pytest` is if you need to keep the main dir on the PYTHONPATH

[the remark this answers is not in the document]
Comment (ric-evans) [reply .github/workflows/wipac-cicd.yml]: okay, I was just about to ask why `-m`

**Pull request merged**